components/dashboard/src/service/json-rpc-user-client.ts

```diff
@@ -33,16 +38,174 @@ export class JsonRpcUserClient implements PromiseClient<typeof UserService> {
+            }
+            if (!!lastUpdatedDetailsNudge) {
+                update.additionalData.profile.lastUpdatedDetailsNudge = lastUpdatedDetailsNudge;
+            }
+            if (!!onboardedTimestamp) {
+                update.additionalData.profile.onboardedTimestamp = onboardedTimestamp;
+            }
+            if (!!signupGoalsOther) {
+                update.additionalData.profile.signupGoalsOther = signupGoalsOther;
+            }
+        }
+
+        if (!!workspaceTimeoutSettings) {
+            update.additionalData.workspaceTimeout = converter.toDurationString(workspaceTimeoutSettings.inactivity);
+            update.additionalData.disabledClosedTimeout = workspaceTimeoutSettings.disabledDisconnected;
+        }
+
+        const user = await getGitpodService().server.updateLoggedInUser(update);
+        return new UpdateUserResponse({
+            user: converter.toUser(user),
+        });
     }
 
     async setWorkspaceAutoStartOptions(
         request: PartialMessage<SetWorkspaceAutoStartOptionsRequest>,
     ): Promise<SetWorkspaceAutoStartOptionsResponse> {
-        throw new ApplicationError(ErrorCodes.UNIMPLEMENTED, "not implemented");
+        const { userId, workspaceAutostartOptions } = request;
+        if (!userId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "userId is required");
+        }
+        if (!workspaceAutostartOptions) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "workspaceAutostartOptions is required");
+        }
+
+        const current = await getGitpodService().server.getLoggedInUser();
+        const currentAdditionalData = { ...current.additionalData };
+        const update: UpdateCurrentUserArgs = {
+            id: userId,
+            fullName: current.fullName!,
+            additionalData: {
+                ...(currentAdditionalData || {}),
+            },
+        };
+        update.additionalData.workspaceAutostartOptions = workspaceAutostartOptions.map((o) =>
+            converter.fromWorkspaceAutostartOption(
+                new User_WorkspaceAutostartOption({
+                    cloneUrl: o.cloneUrl,
+                    editorSettings: o.editorSettings,
+                    organizationId: o.organizationId,
+                    region: o.region,
+                    workspaceClass: o.workspaceClass,
+                }),
+            ),
+        );
+        const user = await getGitpodService().server.updateLoggedInUser(update);
+        return new UpdateUserResponse({
+            user: converter.toUser(user),
+        });
     }
 
-    async deleteUser(request: PartialMessage<DeleteUserRequest>): Promise<DeleteUserResponse> {
-        throw new ApplicationError(ErrorCodes.UNIMPLEMENTED, "not implemented");
+    async deleteUser({ userId }: PartialMessage<DeleteUserRequest>): Promise<DeleteUserResponse> {
+        if (!userId) {
+            throw new ApplicationError(ErrorCodes.BAD_REQUEST, "userId is required");
```

Comment:
It seems dangerous to me, i.e. I found this public-api and trying to delete some other random user's account turn out it deletes my account (XD)

Reply:
But that's the shim of Dashboard, there is no other client that could access it. Do I miss something?

Reply:
Nothing, just our proto define will confuse users in the future

Reply:
But the RPC is like `DeleteUser(targetUserId)`, just in case of shim there is no good corresponding RPC in WebSocket API. 

For the actual API this wont delete the authenticated user's account, but the target.